app/src/androidTest/java/com/amaze/filemanager/asynchronous/services/ftp/FtpServiceEspressoTest.java

```diff
@@ -60,6 +60,7 @@ public void testFtpService() throws IOException {
         PreferenceManager.getDefaultSharedPreferences(service).edit()
                 .putBoolean(FtpService.KEY_PREFERENCE_SECURE, false).commit();
         service.onStartCommand(new Intent(FtpService.ACTION_START_FTPSERVER)
+                .setPackage("com.amaze.filemanager")
```

Comment:
Can't malicious app set the package like its set here?

Reply:
Of course. Just for a first line of defense, hence I called PackageManager to see if the app's package is actually installed on the device.

On the second phase user will be prompted if he/she should allow an external app to switch on/off the FTP server.

Reply:
> Of course. It's just for a first line of defense, hence I called PackageManager to see if the app's package is actually installed on the device.
> (EDIT: OK I get this - malicious app can simply declare intent is sent from Amaze itself. But is there a better solution?)

This is not a viable solution, as it only makes starting the server more difficult and is easily bypassed by malicious actors.

> On the second phase user will be prompted if he/she should allow an external app to switch on/off the FTP server.

This seems like the only viable defence right now, IMO.

Reply:
And `packageName` actually does the opposite thing that makes the intent to fire to the source package itself... :( will work on it.